ddmrp_product_replace/wizards/ddmrp_product_replace.py

```diff
@@ -123,15 +123,18 @@ def _do_replacement_use_existing(self):
             "type": "ir.actions.act_window",
         }
 
+    def _get_new_buffer_default_value(self, replaced_buffer):
+        return dict(
+            product_id=self.new_product_id.id,
```

Comment:
Thanks for improving the readability by moving this part to a separate method. You should use the `replaced_buffer` instead of `self` if I am not mistaken.
```suggestion
            product_id=replaced_buffer.new_product_id.id,
```

Reply:
We should use `self` because the field `new_product_id` is from `ddmrp.product.replace` model, not `stock.buffer`. But I have removed the `replaced_buffer` parameter because is not needed in the function.

Reply:
You are right, thank you for the explanation. So yes, the `replaced_buffer` was not needed indeed (and even misleading apparently) in such a case.